src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -1,57 +1,85 @@
 import React from 'react';
 import { withStyles } from '@material-ui/core/styles';
 
+const topAligned = {
+  '--trans-y': 'var(--tip-size)',
+  '&::before': { bottom: '100%', borderBottomColor: 'var(--bg-color)' },
+  '&.xright': {
+    '--trans-x': 'calc(-100% + 2 * var(--tip-size))',
+    '&::before': { right: 'var(--tip-size)' },
+  },
+  '&.xcenter': {
+    '--trans-x': '-50%',
+    '&::before': { left: 'calc(50% - var(--tip-size))' },
+  },
+  '&.xleft': {
+    '--trans-x': 'calc(-2 * var(--tip-size))',
+    '&::before': { left: 'var(--tip-size)' },
+  },
+};
 const styles = {
   tooltip: {
+    '--bg-color': 'rgba(0, 0, 0, .8)',
+    '--tip-size': '6px',
     position: 'absolute',
     padding: '6px',
-    background: 'rgba(0, 0, 0, .8)',
+    background: 'var(--bg-color)',
     color: 'white',
     borderRadius: '4px',
     pointerEvents: 'none',
+    transform: 'translate(var(--trans-x), var(--trans-y))',
+    '& a': {
+      pointerEvents: 'auto',
+    },
+    '&::before': {
+      position: 'absolute',
+      content: '""',
+      borderWidth: 'var(--tip-size)',
+      borderStyle: 'solid',
+      borderColor: 'transparent',
+    },
+    '&.ytop': {
+      ...topAligned,
+    },
+    '&.ycenter': {
+      '--trans-y': '-50%',
+      '&::before': { top: 'calc(50% - var(--tip-size))' },
+      '&.xright': {
+        '--trans-x': 'calc(-1 * (100% + var(--tip-size)))',
+        '&::before': { left: '100%', borderLeftColor: 'var(--bg-color)' },
+      },
+      '&.xleft': {
+        '--trans-x': 'var(--tip-size)',
+        '&::before': { right: '100%', borderRightColor: 'var(--bg-color)' },
+      },
+    },
+    '&.ybottom': {
+      ...topAligned,
+      '--trans-y': 'calc(-100% - var(--tip-size))',
+      '&::before': { top: '100%', borderTopColor: 'var(--bg-color)' },
+    },
   },
```

Comment:
You probably noticed that I used 'ytop, ycenter, ytop, xleft, xcenter, xright'. Why not just 'top, center, etc'? Mainly because we have two 'center' for horizontal and vertical position, and they could overwrite each other. 

And having 'top, ycenter, bottom' could give us a little of trouble with how we generate the class names at a lower part of the code. Arguable decision.